src/auth/auth.ts

```diff
@@ -100,8 +101,14 @@ export const sendAuthTokenDesyncMetric = () => {
   Ajax().Metrics.captureEvent(Events.user.authToken.desync, {});
 };
 
-export const signOut = (cause: SignOutCause = 'unspecified'): void => {
-  sendSignOutMetrics(cause);
+// This can be called with an expired token, be careful of making any API calls here
+// Any API calls that rely on tokens should fail silently
+export const signOut = async (cause: SignOutCause = 'unspecified') => {
+  // The underlying API calls are already wrapped with `withErrorIgnoring`, but we include it here too to be defensive
+  // It is easier to test this way, and if the underlying API call were to remove that it,
+  // it would cause a serious bug that is hard for a developer to find/test when the token expires after 24 hours.
+  await sendSignOutMetrics(cause).catch(_.noop);
```

Comment:
not super happy about using `.catch(_.noop)` over `withErrorIgnoring`, but `withErrorIgnoring` does some really weird hard to test things. Did not seem worth my time wrangling and mocking a function that has about 4 layers of currying.